OlinkAnalyze/R/bridgeable_plot.R

```diff
@@ -0,0 +1,294 @@
+      dplyr::filter(Project %in% platforms[1],
+                    OlinkID %in% id)
+
+    data2 <- data |>
+      dplyr::filter(Project %in% platforms[2],
+                    OlinkID %in% id)
+
+    data_wider <- data1 |>
+      dplyr::select(SampleID, Assay, Count, OlinkID, UniProt, NPX,
+                    .data[["textcol"]]) |>
+      dplyr::inner_join(data2 |>
+                          dplyr::select(SampleID, Assay, Count, OlinkID,
+                                        UniProt, NPX, .data[["textcol"]]),
+                        by = c("SampleID", "OlinkID", "Assay", "UniProt",
+                               "textcol"),
+                        suffix = c(platforms[1], platforms[2]))
+
+    axis_names <- grep("NPX", colnames(data_wider), value = TRUE)
+
+    r2_lm <- data_wider |>
+      dplyr::mutate(r2_lm = stats::cor(x = .data[[axis_names[1]]],
+                                       y = .data[[axis_names[2]]],
+                                       use = "everything",
+                                       method = "pearson") ^ 2) |>
+      dplyr::pull(r2_lm) |>
+      unique()
+    r2_lm <- signif(r2_lm, 2)
+
+    if (r2_lm < 0.2)
+      caps <- paste0("Possibly bridging background to background")
+    else caps <- ""
+
+    col <- data_wider |>
+      dplyr::filter(OlinkID %in% id) |>
+      dplyr::pull(.data[["textcol"]]) |>
+      unique()
+
+    r2_plt <- data_wider |>
+      ggplot2::ggplot(ggplot2::aes(x = .data[[axis_names[1]]],
+                                   y = .data[[axis_names[2]]])) +
+      ggplot2::geom_point(color = "blue", alpha = 0.4) +
+      ggplot2::geom_smooth(method = "lm", color = "black") +
+      ggpubr::stat_cor(method = "pearson",
+                       ggplot2::aes(label =
+                                      ggplot2::after_stat(.data[["rr.label"]])),
+                       geom = "label") +
+      OlinkAnalyze::set_plot_theme(font = "") +
+      OlinkAnalyze::olink_color_discrete() +
+      ggplot2::facet_wrap(. ~ paste(Assay, OlinkID, sep = "\n"),
+                          scales = "free") +
+      ggplot2::theme(strip.text = ggplot2::element_text(colour = col)) +
+      ggplot2::labs(caption = caps)
+    return(r2_plt)
+  }
+
+  # Median counts plot ----
+  counts_plt <- function(data = data, median_counts_threshold,
+                         id = id) {
+
+    col <- data |>
+      dplyr::filter(OlinkID %in% id) |>
+      dplyr::pull(.data[["textcol"]]) |>
+      unique()
+
+    counts_plt <- data |>
+      dplyr::filter(OlinkID %in% id) |>
+      dplyr::group_by(OlinkID, Project) |>
+      dplyr::mutate(median_count = median(Count, na.rm = TRUE)) |>
+      dplyr::ungroup() |>
+      dplyr::mutate(Assay = factor(paste(Assay, OlinkID, sep = "\n")),
+                    Project = factor(Project)) |>
+      dplyr::group_by(Assay, Project) |>
+      dplyr::reframe(n = .data[["median_count"]]) |>
+      dplyr::ungroup() |>
+      ggplot2::ggplot(ggplot2::aes(x = Assay, fill = Project, y = n,
```

Comment:
would it be worth adding a line after 192 with dplyr::distinct() ? Within each assay we will get multiple lines for each sample within each project so this could help make the function more efficient. If we add distinct we will have 2 lines - 1 for each project

Reply:
Fixed the repeating sample ids by incorporating distinct